figures/2.figure2/scripts/figure2.r

```diff
@@ -0,0 +1,373 @@
+    )
+head(cytokine_values_melted_agg)
+
+# set plot size
+options(repr.plot.width=10, repr.plot.height=5)
+# make the bar plots for the cytokine values
+cytokine_bar_plot <- (
+    ggplot(data=cytokine_values_melted_agg,
+           aes(x=oneb_Metadata_Treatment_Dose_Inhibitor_Dose,
+                y=mean))
+    + geom_bar(stat="identity", position="dodge", color='black', fill='#980CAB')
+    + theme_bw()
+    + geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.2, position=position_dodge(.9))
+    + facet_wrap(.~Cytokine, nrow =1)
+    + theme(axis.text.x = element_text(angle = 90, hjust = 1))
+    + ylab("Cytokine Abundacne")
+    + xlab("Treatment")
+
+)
+
+cytokine_bar_plot
+
+
+# Load data
+data_path_cytokine_values_melted <- file.path("..","..","2.Nomic_nELISA_Analysis/0.Exploratory_Analysis/PBMC/results/PBMC_all_cytokine_values_per_treatment_per_well_melted.csv")
+cytokine_values_melted <- read.csv(data_path_cytokine_values_melted, header = TRUE, sep = ",")
+# filter out the treatments that are not in the list
+cytokine_values_melted <- cytokine_values_melted[cytokine_values_melted$oneb_Metadata_Treatment_Dose_Inhibitor_Dose %in% list_of_treatments,]
+# heatmap of the cytokine values
+# get the aggregated values for the cytokine values across the treatments
+cytokine_values_melted_agg <- cytokine_values_melted %>%
+    group_by(oneb_Metadata_Treatment_Dose_Inhibitor_Dose, Cytokine) %>%
+    summarise(Cytokine_Value = mean(Cytokine_Value))
+
+
+
+# drop two columns by name
+cytokine_values_agg <- subset(cytokine_values, select = -c(fourb_Metadata_Treatment_Dose_Inhibitor_Dose, Metadata_position_x))
+# aggregate the cytokine values across the treatments and cytokine
+cytokine_values_agg <- cytokine_values_agg %>%
+    group_by(oneb_Metadata_Treatment_Dose_Inhibitor_Dose) %>%
+    summarise_all(mean)
+# cytokine_values_agg <- as.matrix(cytokine_values_agg)
+
+head(cytokine_values_melted_agg)
+
+# create a matrix of the cytokine values for the clustering and heatmap
+# unmelt the data
+cytokine_values_agg <- reshape2::dcast(cytokine_values_melted_agg, oneb_Metadata_Treatment_Dose_Inhibitor_Dose ~ Cytokine, value.var = "Cytokine_Value")# make oneb_Metadata_Treatment_Dose_Inhibitor_Dose the rownames
+rownames(cytokine_values_agg) <- cytokine_values_agg$oneb_Metadata_Treatment_Dose_Inhibitor_Dose
+# drop the column
+cytokine_values_agg <- subset(cytokine_values_agg, select = -c(oneb_Metadata_Treatment_Dose_Inhibitor_Dose))
+# convert to matrix
+# cytokine_values_agg <- as.matrix(cytokine_values_agg)
+#
+
+cytokine_values_agg
+
+row_dend <- as.dendrogram(hclust(dist(cytokine_values_agg)))
+col_dend <- as.dendrogram(hclust(dist(t(cytokine_values_agg))))
+
+# remove '[NSU]' from the column names
+colnames(cytokine_values_agg) <- gsub("\\[NSU\\]", "", colnames(cytokine_values_agg))
+
+
+cytokine_values_agg <- as.matrix(cytokine_values_agg)
+
+# set plot size
+options(repr.plot.width=25, repr.plot.height=5)
+heatmap_plot_all <- (
+  Heatmap(
```

Comment:
I am a bit confused here, this looks to be the same code in lines 256-271. Unless both are necessary, I would recommend to not repeat the same code.

Reply:
One is selected cytokines from anova tests and one is all cytokines